File: C/matriz_geral.c
#include <stdio.h>
#include <math.h>

int main() {

    int N, linha, coluna, npositivo;
    double somaPositivo = 0;

    printf("Qual a ordem da matriz? ");
    scanf("%d", &N);

    double mat[N][N];

    for(int i = 0; i < N; i++){
        for(int j = 0; j < N; j++){
            printf("Elemento [%d,%d]: ", i,j);
            scanf("%lf", &mat[i][j]);
        }
    }

    for(int i = 0; i < N; i++){
        for(int j = 0; j < N; j++){
            if(mat[i][j] > 0){
                somaPositivo = somaPositivo + mat[i][j];
            }
        }
    }
    printf("\nSOMA DOS POSITIVOS: %.1lf", somaPositivo);

    printf("\n\nEscolha uma linha: ");
    scanf("%d", &linha);
    printf("LINHA ESCOLHIDA: ");
    for(int j = 0; j < N; j++){
        printf("%.1lf ", mat[linha][j]);
    }

    printf("\n");
    printf("\nEscolha uma coluna: ");
    scanf("%d", &coluna);
    printf("COLUNA ESCOLHIDA: ");
    for(int i = 0; i < N; i++){
        printf("%.1lf ", mat[i][coluna]);
    }

    printf("\n\nDIAGONAL PRINCIPAL: ");
    for(int i = 0 ; i < N; i++){
        printf("%.1lf ", mat[i][i]);
    }

    printf("\n\nMATRIZ ALTERADA:\n");
    for(int i = 0; i < N; i++){
        for(int j = 0; j < N; j++){
            if(mat[i][j] < 0){
                npositivo = pow(mat[i][j], 2);
                mat[i][j] = npositivo;
            }
            printf("%.1lf ", mat[i][j]);
        }
        printf("\n");
    }


    return 0;
}

File: java/matriz_geral.java
import java.util.Locale;
import java.util.Scanner;

public class matriz_geral {

	public static void main(String[] args) {
		
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		int N, l, c;
		double somaMatriz = 0;
		
		System.out.print("Qual a ordem da matriz? ");
		N = sc.nextInt();
		
		double[][] matriz = new double[N][N];
		
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
				System.out.print("Elemento [" + i + "," + j + "]: ");
				matriz[i][j] = sc.nextDouble();
			}
		}
		
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
				if (matriz[i][j] > 0) {
					somaMatriz = somaMatriz + matriz[i][j];
				}
			}
		}
		System.out.println();
		System.out.println("SOMA DOS POSITIVOS: " + String.format("%.1f", somaMatriz));
		
		System.out.println();
		System.out.println();
		System.out.print("Escolha uma linha: ");
		l = sc.nextInt();
		
		System.out.print("LINHA ESCOLHIDA: ");
		for (int j = 0; j < N; j++) {
			System.out.print(matriz[l][j] + " ");
		}
		
		System.out.println();
		System.out.println();
		System.out.print("Escolha uma coluna: ");
		c = sc.nextInt();
		
		System.out.print("LINHA ESCOLHIDA: ");
		for (int i = 0; i < N; i++) {
			System.out.print(matriz[i][c] + " ");
		}
		
		System.out.println();
		System.out.println();
		System.out.print("DIAGONAL PRINCIPAL: ");
		for (int i = 0; i < N; i++) {
			System.out.print(matriz[i][i] + " ");
		}
		
		System.out.println();
		System.out.println();
		System.out.println("MATRIZ ALTERADA: ");
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
				if (matriz[i][j] < 0) {
					matriz[i][j] = Math.pow(matriz[i][j], 2);
				}
				System.out.print(matriz[i][j] + " ");
			}
			System.out.println();
		}
		
		sc.close();
	}

}
